concert/app/test/on_event_stage_changed/test_on_event_stage_changed.py

```diff
@@ -11,83 +11,232 @@
 SUCCESS_RESPONSE = {"ResponseMetadata": {"HTTPStatusCode": 200}}
 
 
-def get_event(stageId="STAGE_CHOICE_COLOR_EMOTION", otherAttributes=None):
-    data = {"stageId": stageId}
-    if otherAttributes:
-        for key, value in otherAttributes.items():
-            data[key] = value
-    return {"body": json.dumps({"event": "EVENT_STAGE_CHANGED", "data": (data)})}
-
-
-def test_event_stage_changed():
-    stageId = "STAGE_CHOICE_COLOR_EMOTION"
-    otherAttributes = {
-        "startTime": "2019-05-14T21:20:03.000Z",
-        "endTime": "2019-05-14T21:21:33.000Z",
-        "frequency": 20,
-    }
-    event = get_event(stageId, otherAttributes)
-
-    expectedMessage = json.dumps(
-        {
-            "event": "EVENT_STAGE_CHANGED",
-            "data": {
-                "stageId": stageId,
-                "startTime": otherAttributes["startTime"],
-                "endTime": otherAttributes["endTime"],
-                "frequency": otherAttributes["frequency"],
-            },
-        }
-    )
+def get_event(
+    stage_id="STAGE_CHOICE_COLOR_EMOTION",
+    display_name=None,
+    start_time=None,
+    end_time=None,
+    frequency=None,
+    form_url=None,
+):
+    data = {"stageId": stage_id}
+    if display_name:
+        data["displayName"] = display_name
+    if start_time:
+        data["startTime"] = start_time
+    if end_time:
+        data["endTime"] = end_time
+    if frequency:
+        data["frequency"] = frequency
+    if form_url:
+        data["formUrl"] = form_url
+    return {"body": json.dumps({"event": "EVENT_STAGE_CHANGED", "data": data})}
```

Comment:
From what I can tell, `event["data"]` isn't being used. Is that right?

Reply:
Not sure I understand what you're asking, so going to answer both ways I could see the question being intended.

If asking about whether the shape here matches the shape of the event that the Lambda will receive as input:
Yes, I think so. The websocket message gets passed in the `"body"` field of the event sent to the Lambda by API Gateway, so the fields we care about are at `json.loads(event["body"])["data"]` rather than `event["data"]`.

If asking about whether these fields are used for the tests:
The `test_event_stage_and_song_list_updated_on_event_stage_change` test case tests that the values in the `data` dict created here are properly written to Dynamo. (The `test_callback_message_published_on_event_stage_change` doesn't do any assertions on static content, but I can go ahead and update it to do so.)

If asking about some other thing:
Sorry for the misunderstanding, and please let me know!

Reply:
I was misreading the code. I thought the `event` dict had both a `body` and `data` key, rather than `data` being nested under `body`. My bad.